File: src/FamilyTree.Api/ClientApp/src/types/note.ts
export interface NoteDto {
  id: string;
  content: string;
  worldId?: string;
  entityId?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateNoteDto {
  content: string;
  worldId?: string;
  entityId?: string;
}

File: src/FamilyTree.Domain/Entities/Note.cs
using FamilyTree.Domain.Common;

namespace FamilyTree.Domain.Entities;

public class Note : BaseEntity
{
    public required string Content { get; set; }
    public Guid? WorldId { get; set; }
    public Guid? EntityId { get; set; }

    public World? World { get; set; }
    public Entity? Entity { get; set; }
}
